x/msgfee/antispam_fee_decorator_test.go

```diff
@@ -0,0 +1,118 @@
+package msgfee_test
+
+import (
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/weavetest"
+	"github.com/iov-one/weave/x/msgfee"
+)
+
+func TestNewAntispamFeeDecorator(t *testing.T) {
+	cases := map[string]struct {
+		ReqFee         coin.Coin
+		AntiSpamFee    coin.Coin
+		Tx             weave.Tx
+		Handler        *weavetest.Handler
+		CheckErr       *errors.Error
+		DeliverErr     *errors.Error
+		WantCheckErr   *errors.Error
+		WantDeliverErr *errors.Error
+		WantCheckFee   coin.Coin
+	}{
+		"anti-spam fee is less than initial fee": {
+			ReqFee:       coin.NewCoin(0, 1234, "DOGE"),
+			Handler:      &weavetest.Handler{},
+			Tx:           &weavetest.Tx{Msg: &weavetest.Msg{RoutePath: "foo/bar"}},
+			WantCheckFee: coin.NewCoin(0, 1234, "DOGE"),
+			AntiSpamFee:  coin.NewCoin(0, 1233, "DOGE"),
+		},
+		"anti-spam fee is equal to initial fee": {
+			ReqFee:       coin.NewCoin(0, 1234, "DOGE"),
+			Handler:      &weavetest.Handler{},
+			Tx:           &weavetest.Tx{Msg: &weavetest.Msg{RoutePath: "foo/bar"}},
+			WantCheckFee: coin.NewCoin(0, 1234, "DOGE"),
+			AntiSpamFee:  coin.NewCoin(0, 1234, "DOGE"),
+		},
+		"anti-spam fee is more than initial fee": {
+			ReqFee:       coin.NewCoin(0, 1234, "DOGE"),
+			Handler:      &weavetest.Handler{},
+			Tx:           &weavetest.Tx{Msg: &weavetest.Msg{RoutePath: "foo/bar"}},
+			WantCheckFee: coin.NewCoin(0, 1235, "DOGE"),
+			AntiSpamFee:  coin.NewCoin(0, 1235, "DOGE"),
+		},
+		"anti-spam fee is zero": {
```

Comment:
Can you add two more "zero" test cases?

Where `AntiSpamFee: coin.Coin{}` and `ReqFee: coin.Coin{}` (only one of the two set to zero in each case).
This should just use the non-zero one... but I want to make sure the ticker "" doesn't cause problems in either case.